Removed duplicate SEPARATOR Constant, cleanup of location tests

pstorch opened the pull request:
Removed duplicate Constant `SEPARATOR`.


I first thought about pushing this small improvement directly to master, but decided otherwise.

Not sure about usage of the Constant in `GeoPointFormatterTest`. I replace the hard coded middle dot with the Constant, to be consistent with the other two test methods. But sometimes its better for Unittests to use redundant hard coded values. To ring the bell when someone accidentally changes the Constant.
What do you think?

Bananeweizen commented:
I'd rather leave the hardcoded literal in the unit tests for multiple reasons:
* Generally, every method under test should be a black box for the test. So the unit test should not be coupled to internals of the class being tested.
* If we were really doing TDD (I admit, I also fail rather often there), it were actually impossible to have the separator constant in the test class, since the assertion would be written first.
* If you feel that this might now lead to code duplication in test methods (i.e. when testing 20 different cases of coordinate formatting), then the better way would be to write one new assertion method assertSomeKindOfFormatting, which contains this (hardcoded) literal just once, and to call that new assertion 20 times.

Independent of that question, I personally don't like the format string instead of the concatenation here. This reduced readability quite a bit, while I see no additional benefit (even though this might save some operations on the CPU, that is neglectable, since this performance difference is not notable by any user). If you don't feel convinced by this, have a look at GCConstants, and find that some of the very huge regular expressions are in fact written as multiple concatenated literals, to at least give some more structure to them.

pstorch commented:
@Bananeweizen I agree with Tests using hard coded and probably duplicated values. That was my thought after I changed it the other way around.

My point with String.format was not performance, but consistency. We mix two methods here to produce a String. But I can see your point in readability, too.

pstorch commented:
Ok, to not keep this open too long, I reverted the String concatenation to keep the readability.
I still removed the duplicate SEPARATOR Constant.
I removed the usage of the SEPARATOR Constant in the tests to not couple the tests to the internals of the code under test.
I found some usages of `assertEquals()` and changed it to `assertThat()` in the location test package.

pstorch commented:
Something wrong with the CI server? 
`com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: Failed to finalize session : INSTALL_FAILED_INSUFFICIENT_STORAGE`

Bananeweizen commented:
Retest this please is the comment to trigger a rebuild

pstorch commented:
Still failing, but this time `Failed to establish session`. Something wrong with the emulator? Try again?

pstorch commented:
retest this please

pstorch commented:
Now again the `INSTALL_FAILED_INSUFFICIENT_STORAGE` error.
Can @mucek4 or @kumy help here?

kumy commented:
not at home right now, please try to deactivate failing executors.

On October 1, 2016 10:37:54 PM GMT+02:00, Peter Storch <notifications@github.com> wrote:
>Now again the `INSTALL_FAILED_INSUFFICIENT_STORAGE` error.
>Can @mucek4 or @kumy help here?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/cgeo/cgeo/pull/6001#issuecomment-250936137

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

samueltardieu commented:
I'm on it.

samueltardieu commented:
retest this please

pstorch commented:
Thanks
